modules/ROOT/pages/field-level-capabilities.adoc

```diff
@@ -0,0 +1,256 @@
+will be applied. The name of the acting parameter must be the same as the name
+of the parameter in the component to take that parameter’s value.
+
+Use the *@Binding* annotation to define an acting parameter. The *@FieldValues*
+annotation and the `@SampleData` annotation can be given an array of bindings
+to define the values the acting parameter must take.
+
+When a binding is defined, it declares the value an acting parameter will
+take by giving an extraction expression. This extraction expression is a
+DataWeave expression that uses only selectors to retrieve the value for the
+acting parameter.
+
+The available context for the expression has the value of all the component's
+parameters. The expression must start with the name of the parameter of the
+component. If the value comes from a part of the parameter, the expression
+follows the selector to the desired field.
+
+The following is an example where a parameter named `body` has a field with
+values that depend on the acting parameter `workspace`:
+
+----
+public class ChatOperations {
+ public void sendChatMessage(@Connection ChatConnection chatConnection,
+       @FieldValues(value = ChannelsValueProvider.class, targetSelectors = "channelId",
+     bindings = {@Binding(actingParameter = "workspace",  extractionExpression = "body.routingInfo.workspace")}))
+       @TypeResolver(ChatMessageResolver.class) @Content InputStream body) {
+    // Do message sending
+  }
+}
+----
+
+This acting parameter is taken from the same `body` parameter.
+
+The following shows the value provider used in this example:
+
+----
+public class ChannelsValueProvider implements ValueProvider {
+
+ @Connection
+ private ChatConnection chatConnection;
+
+ @Parameter
+ private String workspace;
+
+ @Override
+ public Set<Value> resolve() {
+   return ValueBuilder.getValuesFor(chatConnection.getChannels(workspace));
+ }
+
+ @Override
+ public String getId() {
+   return "Channels value provider";
+ }
+}
+----
+
+The `workspace` acting parameter defined in the value provider is bound to the
+extraction expression that takes the value of the field from the context
+of the component.
+
+== Sample Data Example
+
+The following is an example of a `@SampleData` annotation that binds an acting
+parameter to the value of a field of a parameter in the component:
+
+----
+public class ChatOperations {
+@SampleData(value = ChatMessageSampleDatatProvider.class, bindings = {@Binding(actingParameter = "workspace",  extractionExpression = "body.routingInfo.workspace")}))
+ public Result<InputStream, ChatAttributes> retrieveChatMessage(@Connection ChatConnection chatConnection,
+         @TypeResolver(ChatMessageResolver.class) @Content InputStream body) {
+    // Do chat message retrieval
+  }
+}
+----
+
+In the *ChatMessageSampleDatatProvider*, there is a parameter named `workspace`
+that provides the sample data. The value of this acting parameter
+is taken from the extraction expression defined in the binding.
+
```

Comment:
How do the next three topics relate to all the previous? Seems like maybe the types of subtopics should be further grouped.

Reply:
Do they all relate to "acting parameters"; do we need a subtopic further defining acting parameters near the topic set intro?